File: app/src/main/java/game/shenle/com/dragger/AppComponent.java
/*
 * Copyright (C) 2017 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package game.shenle.com.dragger;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import game.shenle.com.MyApplication;
import game.shenle.com.dragger.module.ActivityBuildersModule;
import game.shenle.com.dragger.module.AppModule;

@Singleton
@Component(modules = {
        AndroidInjectionModule.class,
        AppModule.class,
        ActivityBuildersModule.class})
public interface AppComponent {
     void inject(MyApplication myApplication);
}

File: app/src/main/java/game/shenle/com/dragger/AppComponent.kt
package game.shenle.com.dragger

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import game.shenle.com.MyApplication
import game.shenle.com.dragger.module.AppModule
import game.shenle.com.dragger.module.MainActivityModule
import javax.inject.Singleton

/**
 * Created by shenle on 2017/11/16.
 */
@Singleton
@Component(modules = arrayOf(AppModule::class,
        AndroidInjectionModule::class,
        MainActivityModule::class))
interface AppComponent {
//    @Component.Builder
//    interface Builder {
//        @BindsInstance
//        fun application(application: MyApplication): Builder
//        fun build(): AppComponent
//    }
    fun inject(application: MyApplication)
}